packages/core/src/index.ts

```diff
@@ -1,6 +1,7 @@
 // reflect-metadata used for class-transformer + class-validator
 import 'reflect-metadata'
 
+export { AgentContext } from './agent/AgentContext'
```

Comment:
Why do we want to expose the `AgentContext` publicly?

Reply:
If we want to inject it (in extension modules for example) we need it. So it's not necessary for the public api for users, but is needed if you want to write extension modules.